### PR TITLE
TMA-711: segments filter works correctly in users brick

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 1]
```diff
@@ -170,7 +170,7 @@ def call(params)
                       domain_clients = domain.clients(:all, data_product)
                       if params.segments
                         segment_uris = params.segments.map(&:uri)
-                        domain_clients.select! { |c| segment_uris.include?(c.segment_uri) }
+                        domain_clients = domain_clients.select { |c| segment_uris.include?(c.segment_uri) }
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'c.segment_uri' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Calls 'segment_uris.include?(c.segment_uri)' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: Has the variable name 'c' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/607/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/408/ : SUCCESS in 31s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/716/ : SUCCESS in 31s

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/608/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1144/ : SUCCESS in 47m 55s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/902/ : SUCCESS in 30m 35s

**Pull request merged**